packages/sommelier.rb

```diff
@@ -2,106 +2,256 @@
+SOMMWPROCS=\$(pgrep -fa \"sommelier.elf --master\" 2> /dev/null)
+SOMMXPIDS=\$(pgrep -f \"sommelier.elf -X\" 2> /dev/null)
+SOMMXPROCS=\$(pgrep -fa \"sommelier.elf -X\" 2> /dev/null)
+if checksommelierwayland && checksommelierxwayland ; then
+  echo -e \"sommelier processes running: \$(echo \$SOMMWPIDS \$SOMMXPIDS)\"
+else
+  echo \"some sommelier processes failed to start\"
+  echo -e \"sommelier processes running: \$SOMMWPROCS \\n \$SOMMXPROCS\"
+  exit 1
+fi
+EOF"
+
+      # stopsommelier
+      system "cat <<'EOF'> stopsommelier
+#!/bin/bash
+SOMM=\$(pgrep -fc sommelier.elf 2> /dev/null)
+if [[ \"$SOMM\" -gt \"0\" ]]; then
+  pkill -f sommelier.elf &>/dev/null
+  pkill -F #{CREW_PREFIX}/var/run/sommelier-wayland.pid &>/dev/null
+  pkill -F #{CREW_PREFIX}/var/run/sommelier-xwayland.pid &>/dev/null
+else
+  exit 0
+fi
+if [[ \"$(pgrep -fc sommelier.elf 2> /dev/null)\" -gt \"0\" ]]; then
+  echo \"sommelier failed to stop\"
+  exit 1
+else
+  echo \"sommelier stopped\"
+fi
+EOF"
+
+      # restartsommelier
+      system "echo '#!/bin/bash' > restartsommelier"
+      system "echo 'stopsommelier && startsommelier' >> restartsommelier"
+      end
     end
   end
-
+    
   def self.install
-    Dir.chdir ("sommelier") do
-      system "make", "PREFIX=#{CREW_PREFIX}", "SYSCONFDIR=#{CREW_PREFIX}/etc", "DESTDIR=#{CREW_DEST_DIR}", "install"
-      system "install -Dm755 sommelierd #{CREW_DEST_PREFIX}/sbin/sommelierd"
-      system "install -Dm755 initsommelier #{CREW_DEST_PREFIX}/bin/initsommelier"
-      system "install -Dm755 stopsommelier #{CREW_DEST_PREFIX}/bin/stopsommelier"
-      system "install -Dm755 startsommelier #{CREW_DEST_PREFIX}/bin/startsommelier"
-      system "install -Dm644 .sommelier.env #{CREW_DEST_HOME}/.sommelier.env"
+    Dir.chdir ("vm_tools/sommelier") do
+      system "DESTDIR=#{CREW_DEST_DIR} ninja -C build install"
+      Dir.chdir ("build") do
+        FileUtils.mv "#{CREW_DEST_PREFIX}/bin/sommelier", "#{CREW_DEST_PREFIX}/bin/sommelier.elf"
+        system "install -Dm755 sommelier_sh #{CREW_DEST_PREFIX}/bin/sommelier"
+        system "install -Dm755 sommelierd #{CREW_DEST_PREFIX}/sbin/sommelierd"
+        system "install -Dm755 startsommelier #{CREW_DEST_PREFIX}/bin/startsommelier"
+        FileUtils.ln_sf 'startsommelier', "#{CREW_DEST_PREFIX}/bin/initsommelier"
+        system "install -Dm755 stopsommelier #{CREW_DEST_PREFIX}/bin/stopsommelier"
+        system "install -Dm755 restartsommelier #{CREW_DEST_PREFIX}/bin/restartsommelier"
+        system "install -Dm755 sommelierrc #{CREW_DEST_PREFIX}/etc/sommelierrc"
+        system "install -Dm644 .sommelier.env #{CREW_DEST_HOME}/.sommelier.env"
+
+      end
     end
   end
-
+  
   def self.postinstall
     puts
     puts "To complete the installation, execute the following:".lightblue
-    puts "echo '# Sommelier environment variables + daemon' >> ~/.bashrc".lightblue
+    puts "echo '# Sommelier envronment variables + daemon' >> ~/.bashrc".lightblue
```

Comment:
It was spelled correctly the first time. :)

Reply:
I'm not going to go into how much time I wasted trying to get gtkspell working with geany yesterday...

Reply:
Fixed.